### PR TITLE
Add jitter to sleeps

### PR DESCRIPTION
**Pull request by paulocsanz**
- Also add test to ensure ERR messages from faktory don't trigger a
retry

Do we want jitters in heartbeats? And is 5 seconds enough/too much?

### CODE REVIEW

**Review by zacharyhamm — APPROVED** [review]
What if we did:

`let jitter = thread_rng().gen_range(0..=backoff);` (and then just use `jitter` as the sleep)

This is equivalent to what this article calls "Full Jitter": https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/


But this is probably fine enough for our purposes

**Comment by paulocsanz** [comment]
Nice! Will do.

**Comment by paulocsanz** [comment]
Done, can you double check? @zacharyhamm

**Comment by jhelwig** [comment]
@paulocsanz Good point about the heartbeats. Eventually, it'd probably be good to have it be 15 seconds +/- a jitter amount, so that way they're 15 seconds apart on average, but not always _exactly_ 15 seconds. That said, I'm more than OK with that been addressed later, and not being a blocker for this PR.

File: src/lib.rs [hunk 1]
```diff
@@ -246,23 +245,24 @@ impl Client {
...
                 _ => break,
             }
 
+            let jitter = thread_rng().gen_range(1..=backoff);
             match dbg!(Connection::new(config.clone()).await) {
                 Err(_) => {
                     tokio::select! {
                         _ = shutdown_channel.recv() => break,
-                        _ = tokio::time::sleep(Duration::from_secs(1 << backoff)) => {}
+                        _ = tokio::time::sleep(Duration::from_secs(1 << jitter)) => {}
```
Comment (zacharyhamm) [inline src/lib.rs:265]: 👍

**Review by zacharyhamm — APPROVED** [review]

**Comment by paulocsanz** [comment]
Oh @jhelwig I saw your comment after merging, ok we can deal with that later, I moved to 10 to 20 as I felt the average of 15 was reasonable, but we can change that later as we wish.

**Comment by jhelwig** [comment]
@paulocsanz No worries. You pretty much already did was I was talking about.

**Pull request merged**